### DISCUSSION TITLE
Using a glob in `generates` property of a task

### DISCUSSION THREAD

**Discussion by adamurban98** [go-task/task#446] [Q&A]
Hi there, let me briefly describe my problem:

If I opt for `method: checksum` it is enough for me to provide a `sources` property for a task to prevent it from rebuilding unnecessarily. However, I would still like to set the `generates` property in order to be able to detect manipulation with the generated files (e.g. the user deletes or moves one of them).

The trouble is, that one of my tasks is used to generate a folder with lots of images and this output is nondeterministic. 

## Desired behaviour

If I run `task generate-images` following files will be generated:

```
output_dir/img1.png
output_dir/img3.png
output_dir/img4.png
output_dir/img6.png
...
```
If the `generates` is set to `output_dir` or `output_dir/*` ,then it should remember the output folders checksum and if I run simply `task generate-images` again nothing should happen.
But if I force run it with `task generate-images -f`, or if change some file in the `sources` field, then the task should run again and a new set of images should be generated:
```
output_dir/img10.png
output_dir/img11.png
output_dir/img12.png
output_dir/img16.png
...
```

This behaviour already works if the `generates` property is set to be a list of specific files, while the `sources` property can contain globs as well. As I tried to explain above, however, in my case I cannot list a concrete list of files for the `generates` property as I don't know what those files will be.

## Questions

1. Is it possible to solve this problem by somehow cleverly setting up the `generates` and `sources` attributes of the task?
2. Would you suggest to write a custom checking script? If yes, then how? 
3. Can I somehow use the .CHECKSUM variable? What exactly is that variable? If i have multiple `generates` files, how can I programmatically calculate the same hash?

 Thanks for any answers/suggestions in advance :slightly_smiling_face:

**Comment by andreynering** [comment]
Hi @adamurban98,

1. It's not clear for me why setting `generates` to `output_dir/**/*` does not do what you want. From what I could understand it should.
2. [Writing a script](https://taskfile.dev/#/usage?id=using-programmatic-checks-to-indicate-a-task-is-up-to-date) is a possibility, but if your code generation is not deterministic I'm afraid it won't be possible to solve this use case with this feature.
3. I don't think that `.CHECKSUM` will help with with this use case.

Sorry for not having more concrete answers for you. I think you should try to have a deterministic workflow, if possible.
